lib/spack/spack/test/util/compression.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+import shutil
+
+import pytest
+
+from llnl.util.filesystem import working_dir
+
+from spack.paths import spack_root
+from spack.util import compression as scomp
+from spack.util.executable import CommandNotFoundError
+
+datadir = os.path.join(spack_root, 'lib', 'spack',
+                       'spack', 'test', 'data', 'compression')
+
+ext_archive = {}
+[ext_archive.update({ext: '.'.join(['Foo', ext])}) for
+    ext in scomp.ALLOWED_ARCHIVE_TYPES]
```

Comment:
I think the unit tests are failing because this expects a `Foo.Z` (and we haven't added one, which I think we agreed is fine but we'll have to exclude `Foo.Z` from this list)

Reply:
I just got my VM working, I can create .Z/Tar.Z files again, so I'll upload those now.